### PR TITLE
Reorders code to allow writing PGN cloud in the run phase

### PR DESCRIPTION
**Pull request by singhbalwinder**
PGN was generating its probability cloud only in "generate" phase. This
cloud should be generated in the "run" phase so that baseline blessing
script can copy this file over when we bless baselines.

[BFB] - Bit-For-Bit

### CODE REVIEW

File: cime/scripts/lib/CIME/SystemTests/pgn.py [hunk 1]
```diff
@@ -570,53 +570,57 @@ def run_phase(self):
                 
                 iinst += 1
 
+        #cloud generation
 
-        if(self._case.get_value("GENERATE_BASELINE")):
+        if(debug):
+            print("PGN_INFO: cloud generation-gen base")
```
Comment (jgfouca) [inline cime/scripts/lib/CIME/SystemTests/pgn.py]: logger.debug(...) is what you want here

Comment (jgfouca) [reply cime/scripts/lib/CIME/SystemTests/pgn.py]: CIME already supports --debug flag for all tools.

File: cime/scripts/lib/CIME/SystemTests/pgn.py [hunk 2]
```diff
@@ -60,7 +60,7 @@
 prtstr     = ['woprt','posprt','negprt']
 
 #file name for file containing PGE cloud
-fcld_nc  = 'cloud.nc'
+fcld_nc  = 'PGN_cloud.nc'
```
Comment (jgfouca) [inline cime/scripts/lib/CIME/SystemTests/pgn.py]: global variables should have all-caps names.

File: cime/scripts/lib/CIME/SystemTests/pgn.py (hunk 1 above)
Comment (singhbalwinder) [reply cime/scripts/lib/CIME/SystemTests/pgn.py]: Thanks! I tried `logger.debug` in the past and it didn't work at that time. perhaps I wasn't using it correctly. Would you please point me some place where it is used in the scripts or tests? Is `--debug`  flag a `create_test` scripts's argument?

File: cime/scripts/lib/CIME/SystemTests/pgn.py (hunk 2 above)
Comment (singhbalwinder) [reply cime/scripts/lib/CIME/SystemTests/pgn.py]: Thanks! I didn't know that. I will make that change in this script.

File: cime/scripts/lib/CIME/SystemTests/pgn.py (hunk 1 above)
Comment (jgfouca) [reply cime/scripts/lib/CIME/SystemTests/pgn.py]: @singhbalwinder , all CIME tools should support --debug. You can grep for "logger.debug".

**Comment by singhbalwinder** [comment]
@jgfouca : I think I have addressed all your comments above. Would you please take another look to see if it all looks good to you? Thanks!

**Comment by jgfouca** [comment]
Much better, thx!

**Comment by singhbalwinder** [comment]
Skipping integration testing.

**Pull request merged**